### PR TITLE
Add funcorder linter

### PR DESCRIPTION
**Pull request by shijiesheng**
**What changed?**

We need to sort the order of access methods in preparation for add codegen components. So I wrote a simple ast-based linter that automatically scan through the package and apply the changes. 

**Why?**

**How did you test it?**

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: tools/linter/funcorder/cmd/funcorder/main.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package main
+
+import (
+	"github.com/uber/cadence/tools/linter/funcorder"
+	"golang.org/x/tools/go/analysis/singlechecker"
+)
+
+func main() {
```
Comment (Shaddoll) [inline tools/linter/funcorder/cmd/funcorder/main.go:8]: How is the tool used? Do we need to run it manually? Do we need to pass the file we want to fix?

Comment (shijiesheng) [reply tools/linter/funcorder/cmd/funcorder/main.go:8]: Currently it's a script that runs manually. go run main.go will output the command line.

**Review by Shaddoll — APPROVED** [review]

File: tools/linter/funcorder/cmd/funcorder/main.go (hunk 1 above)
Comment (Groxx) [reply tools/linter/funcorder/cmd/funcorder/main.go:31]: I'm guessing this isn't hooked into the `make lint` step yet?  should that be done?

Comment (Groxx) [reply tools/linter/funcorder/cmd/funcorder/main.go:31]: ah.  and we should probably move this to internal/tools/go.mod

**Pull request merged**